### ISSUE TITLE
MalformedJsonException: Invalid escape sequence

### ISSUE THREAD

**Issue by RuralHunter** [google/gson#2471]
# Gson version

2.7-2.10.1

# Java / Android version

Java 8

# Used tools

- [+] Maven; version: 
- [+] Gradle; version: 
- [ ] ProGuard (attach the configuration file please); version: 
- [ ] ...

# Description

Starting from version 2.7, the code below to parse the json in attachement raises MalformedJsonException. The older versions(I tested 2.4.x and 2.6.x) are fine.
```java
        Gson gson=new Gson();
        JsonReader jsonReader=new JsonReader(new InputStreamReader(GsonTest.class.getResourceAsStream("/badjson.txt")));
        //jsonReader.setLenient(true);
        AjaxResponse o = gson.fromJson(jsonReader,AjaxResponse.class);
        System.out.println(o);
```

## Expected behavior

The parsing should be OK.

## Actual behavior

gson raises MalformedJsonException

# Reproduction steps

See the code above and the attachment json content.

# Exception stack trace

```
Exception in thread "main" com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Invalid escape sequence at line 1 column 2648 path $.content
	at com.google.gson.Gson.fromJson(Gson.java:1241)
	at [com.google.gson.Gson.fromJson(Gson.java:1186)]
```
Attachment: [badjson.txt](https://github.com/google/gson/files/12355336/badjson.txt)

**Comment by eamonnmcmanus** [comment]
Thanks for attaching the problematic input! If we look at the Unicode character with the offset in the error message (2648), we find this in the vicinity:
```
中環醫務所格局及級數的深圳牙科診\所
```
That `\所` at the end is not a valid [JSON escape sequence](https://datatracker.ietf.org/doc/html/rfc8259#section-7) so Gson is right to reject it. It may be that earlier versions of Gson accepted this, but I don't think they should have. If Gson has rejected this input since version 2.7, released in 2016, we're probably not going to start accepting it now.

**Comment by Marcono1234** [comment]
> It may be that earlier versions of Gson accepted this, but I don't think they should have.

Looks like this was fixed by 3f8726ecafe9c40498255eed1d1893fd665e237b (which was a manual merge of #862).

(Just adding this to provide additional information; I agree though with Éamonn that throwing an exception here is the right behavior.)

**Comment by RuralHunter** [comment]
Ok, fair enough. Thanks guys.
